attachments/static/attachments/js/attachments.js

```diff
@@ -33,7 +40,14 @@
                 url: settings.url,
                 data: data,
                 success: function(html) {
+                    var attachmentsRetryButtonHTML = null;
+                    if (settings.enableRetry && $(settings.attachmentsRetryButtonID).length) {
+                        attachmentsRetryButtonHTML = $(settings.attachmentsRetryButtonID)[0].outerHTML;
+                    }
                     $(settings.container).empty().append(html).trigger('table-changed');
+                    if (attachmentsRetryButtonHTML) {
+                        $(settings.container).append(attachmentsRetryButtonHTML);
+                    }
```

Comment:
Can you explain what's going on here in this success function? It looks like if retry is enabled & the button exists, then you're grabbing the outter HTML for that button.

Then, checking if you have the outter HTML and if you do, appending it to the container.  Is that right?

If the button already exists, you shouldn't need to append it anywhere.  If the button exists, I think you just need to hide it.  You only show/add the button in the event of an error

Reply:
When I had originally written this, it was because the retry button was getting removed due to this line: https://github.com/imsweb/django-dynamic-attachments/blob/109a522ac9f67de2ea1b0bd8c5c6ea7100c2ea62/attachments/static/attachments/js/attachments.js#L36
It was a hacky workaround, and had the issues that you just mentioned. There is a better solution now; instead of doing `$(settings.container).append(...)`, we are doing `$(settings.container).after(...)`